src/query_planner/abstract_plan.py

```diff
@@ -1,19 +1,14 @@
-from abc import ABC, abstractmethod
-from enum import IntEnum, unique
+from abc import ABC
 
-
-@unique
-class PlanNodeType(IntEnum):
-    SEQSCAN = 1
-    # add other types
+from src.query_planner.types import PlanNodeType
 
 
 class AbstractPlan(ABC):
 
-    def __init__(self):
+    def __init__(self, node_type):
```

Comment:
what will be the node_type for an abstract scan node? I mean if we don't want to assign any type to abstract nodes then it is waste to keep it as a member variable

Reply:
Having a node_type for every plan makes it easy to perform a condition check (switch cases). If we see the conversion of a plan tree to the execution tree we will have to use the type to determine which executor needs to be created. 

Having this in the superclass (AbstractPlan) will make sure any Plan Node will have this property. 

If you check the implementation of the _build_execution_tree method of PlanExecutor, we make use of this node_type attribute. If we don't have this we need to use **isinstance**, which I feel is not a great thing to do.

Reply:
Agreed we won't be using isinstance, rather I meant making node_type as a method of each plan node. We don't need a variable in the base class. And we won't need to pass the node_type each time to create a node.